### DISCUSSION TITLE
Return decision process also when entity recogniser fails to understand where.

### DISCUSSION THREAD

**Discussion by olivierpaling** [microsoft/presidio#727] [Ideas]
Hi!

Looking at current supported PII entities like IBAN, CreditCard and UK NHS to name some examples, we see a nice combination of pattern recognition with context words and checksums (+ some extra magic). This makes the recognition of such classifiers accurate which is a great thing in itself for the purpose that we are serving. 

But if you turn that around when for instance an IBAN code has 1 incorrect number included in the code (4 instead of 5 e.a) it breaks halfway through and is not returned in the request. 
There is however value in giving back that it's partially correct and/or that the user might have made a mistake in typing an IBAN (active classification, a use case that we serve).

I tried using the ```"return_decision_process": true``` as I expected that it would return that the pattern was found but the checksum would fail or so. But that works in an 'all or nothing' kinda way and only get's returned when IBAN was found in full.

Another way would be to adjust scoring for these cases.

Any thoughts here :) ?

Thanks.
Olivier

**Comment by rickgoud** [comment]
Would be a great improvement indeed! Maybe extend with including 'simple regex' and if simple is successful and complex fails, return likelihood of 75% (or so?)

**Comment by shiranr** [comment]
Hi @olivierpaling, 

Thanks you for reaching out and for the suggestion, currently, Presidio supports only valid numbers of those types. 
I added the request to our backlog and we will consider adding this feature to our future road map.
In the meantime, we welcome community contributions, if you like to have a go at it.

Have a great week,
Shiran.
